### PR TITLE
fix: read AMT and SMB passwords without terminal echo unless requested

### PR DESCRIPTION
**Pull request by Craig-Spencer-12**
* Adds the flag --echo-password to show AMT Password in the terminal
* Default action is now to hide the passwords in the terminal

### CODE REVIEW

File: internal/rps/message.go [hunk 1]
```diff
@@ -165,11 +165,8 @@ func (p Payload) CreateMessageRequest(flags flags.Flags) (Message, error) {
 	if payload.CurrentMode != 0 {
 		if flags.Password == "" {
 			for flags.Password == "" {
-				fmt.Println("Please enter AMT Password: ")
-				// Taking input from user
-				_, err = fmt.Scanln(&flags.Password)
-				if err != nil {
-					return message, err
+				if _, err := flags.ReadPasswordFromUser(); err != nil {
+					return message, errors.New("Missing or incorrect password")
```
Comment (rsdmike) [inline internal/rps/message.go:169]: there should be an existing error for this in the utils package

File: internal/local/deactivate_test.go [hunk 2]
```diff
@@ -24,7 +27,7 @@ func TestDeactivation(t *testing.T) {
 	})
 
 	t.Run("returns UnableToDeactivate when ControlMode is pre-provisioning (0)", func(t *testing.T) {
-		lps := setupService(f)
+		lps := setupService(f2)
```
Comment (rsdmike) [inline internal/local/deactivate_test.go:30]: if this is the only place f2 is used, i woudl suggest moving the creation/allocation of f2 into this test alone as its the only test it is used in. Alternatively, re-use f2 across all tests that need to fail, i.e. TestDeactivateACM. Info_test.go could be evaluated for that as well.

File: internal/flags/flags.go [hunk 3]
```diff
@@ -96,10 +97,12 @@ type Flags struct {
 	SambaService                        smb.ServiceInterface
 	MEBxPassword                        string
 	ConfigTLSInfo                       ConfigTLSInfo
+	pr                                  utils.PasswordReader
```
Comment (rsdmike) [inline internal/flags/flags.go:100]: no other variable in this struct follows this naming convention. For the sake of consistency please use `passwordReader`

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: internal/flags/flags.go (hunk 3 above)
Comment (Craig-Spencer-12) [reply internal/flags/flags.go:100]: done

File: internal/local/deactivate_test.go (hunk 2 above)
Comment (Craig-Spencer-12) [reply internal/local/deactivate_test.go:30]: done

File: internal/rps/message.go (hunk 1 above)
Comment (Craig-Spencer-12) [reply internal/rps/message.go:169]: Done

**Review by rsdmike — APPROVED** [review]
Looks good, can merge for now. However, the behavior of asking the password is different for rps vs local. RPS won't accept a blank password and continues asking until one is given where local just accepts and quits. It would be ideal if the both behaved the same, that is, they both should not allow continuing (and not exit) if no password is entered.

**Pull request merged**